RFCs/polka-index.md

```diff
@@ -0,0 +1,79 @@
+# RFC-001: Polka-index
+
+## Abstract
+
+This document describes Polka-index, a indexing application that maps Content Identifiers (CIDs) to the Peer IDs of storage providers in a peer-to-peer (P2P) network. Users interact with Polka-index through a simple JSON API, while Polka-Index communicates with storage providers over a P2P network using [libp2p](https://docs.libp2p.io/). This document outlines the system's architecture, key components, and communication protocols.
```

Comment:
Should we add the design of the endpoint that we're going to expose?
And what endpoint we need to implement on Storage Provider?

Reply:
I'll add a design for the HTTP endpoint that we need in the indexer node. We do not need to add a HTTP endpoint to the storage provider server, the storage provider server functionality needs to be extended to share deal information with the P2P network.

Reply:
I don't think we need to expose an API in the RFC, if we do, it should be an example and not something "set in stone"